src/parser.ts

```diff
@@ -3892,15 +3891,14 @@ export function parseMemberOrUpdateExpression(
   context: Context,
   expr: ESTree.Expression,
   inGroup: 0 | 1,
+  inChain: 0 | 1,
```

Comment:
Add an argument to parseMemberOrUpdateExpression to track the top level of a chain expression.